### PR TITLE
[UX] Print useful message when image id not found

### LINKED ISSUE #2519: UX: image 'xxx' not found error should print to-fix and cloud command used to query
See https://github.com/run-house/runhouse/issues/100. An error like `ValueError: Image 'pytorch-latest-cpu' not found in GCP.` is not informative enough to help users debug. Let's print something like a to-fix and cloud command used to query.

(Thanks @steve-marmalade for raising this!)

### PR DESCRIPTION
**Pull request by cblmemo**
Fixes #2519 by adding some messages to help the user debug. Most of the messages are pasted from our `yaml-spec.rst`.

**Current docker implementation won't check whether the docker image exists, so no message for docker is added in this PR.**

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
```bash
(sky-dev) ➜  skypilot git:(ux-image-id) $ sky launch --cloud gcp --image-id some-random-string
ValueError: Image 'some-random-string' not found in GCP.

To find GCP images: https://cloud.google.com/compute/docs/images
Format: projects/<project-id>/global/images/<image-name>
Example: projects/deeplearning-platform-release/global/images/common-cpu-v20230615-debian-11-py310

To find machine images: https://cloud.google.com/compute/docs/machine-images
Format: projects/<project-id>/global/machineImages/<machine-image-name>

You can query image id using: gcloud compute images list --project <project-id> --no-standard-images
To query common AI images: gcloud compute images list --project deeplearning-platform-release | less
```
```bash
(sky-dev) ➜  skypilot git:(ux-image-id) $ sky launch --cloud aws --image-id some-random-string --region us-east-1
ValueError: Image 'some-random-string' not found in AWS region us-east-1.

To find AWS AMI IDs: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html#examples
Example: ami-0729d913a335efca7
```
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/gcp.py [hunk 1]
```diff
@@ -83,6 +83,16 @@
 # TODO(zhwu): Move the default AMI size to the catalog instead.
 DEFAULT_GCP_IMAGE_GB = 50
 
+_IMAGE_NOT_FOUND_UX_MESSAGE = (
+    'Image {image_id!r} not found in GCP.\n'
+    '\nTo find GCP images: https://cloud.google.com/compute/docs/images\n'
+    'Format: projects/<project-id>/global/images/<image-name>\n'
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:89]: bright the value?

File: sky/clouds/gcp.py [hunk 2]
```diff
@@ -83,6 +83,16 @@
...
+    'Example: projects/deeplearning-platform-release/global/images/common-cpu-v20230615-debian-11-py310\n'  # pylint: disable=line-too-long
+    '\nOr machine image: https://cloud.google.com/compute/docs/machine-images\n'
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:91]:
```suggestion
    '\nTo find machine images: https://cloud.google.com/compute/docs/machine-images\n'
```

File: sky/clouds/gcp.py [hunk 3]
```diff
@@ -83,6 +83,16 @@
...
+    'Format: projects/<project-id>/global/machineImages/<machine-image-name>\n'
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:92]: bright the value?

File: sky/clouds/gcp.py [hunk 4]
```diff
@@ -83,6 +83,16 @@
...
+    '\nYou can query image id using:\n'
+    'gcloud compute images list --project <project-id> --no-standard-images')
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:94]: bright the value

Add another hint like the below?
```
To query common AI images: <bright>gcloud compute images list --project deeplearning-platform-release | less</bright>
```

File: sky/clouds/aws.py [hunk 5]
```diff
@@ -241,8 +241,10 @@ def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
             return DEFAULT_AMI_GB
         except aws.botocore_exceptions().ClientError:
             with ux_utils.print_exception_no_traceback():
-                raise ValueError(f'Image {image_id!r} not found in '
-                                 f'AWS region {region}') from None
+                raise ValueError(
+                    f'Image {image_id!r} not found in AWS region {region}.\n'
+                    f'\nTo find AWS AMI ids: https://leaherb.com/how-to-find-an-aws-marketplace-ami-image-id\n'  # pylint: disable=line-too-long
```
Comment (concretevitamin) [inline sky/clouds/aws.py:246]:
```suggestion
                    f'\nTo find AWS AMI IDs: https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-images.html#examples\n'  # pylint: disable=line-too-long
```

**Review by concretevitamin — COMMENTED** [review]
Thanks @cblmemo. Minor UX nits. I guess it's fine to leave `docker:xxx` to the future?

**Review by concretevitamin — APPROVED** [review]
Thanks @cblmemo, LGTM.

File: sky/clouds/gcp.py [hunk 6]
```diff
@@ -83,15 +84,18 @@
 # TODO(zhwu): Move the default AMI size to the catalog instead.
 DEFAULT_GCP_IMAGE_GB = 50
 
+USER_PORTS_FIREWALL_RULE_NAME = 'sky-ports-{}'
```
[the remark this answers is not in the document]
Comment (cblmemo) [reply sky/clouds/gcp.py:87]: Sorry wdym by 'short comment'? Did you mean to short the following string so we could get rid of the `# pylint: disable=line-too-long`?

[the remark this answers is not in the document]
Comment (concretevitamin) [reply sky/clouds/gcp.py:87]:
```
# Describe this var briefly.
USER_PORTS_FIREWALL_RULE_NAME = 'sky-ports-{}'
```
:)

**Pull request merged**